File: frontend/types/openaiResponse.ts
export type OpenAIContentItem = {
  type: string;
  annotations?: any[];
  logprobs?: any[];
  text?: string;
};

export type OpenAIOutputMessage = {
  id: string;
  type: string;
  status: string;
  content: OpenAIContentItem[];
  role: string;
};

export type OpenAIReasoning = {
  effort?: any;
  summary?: any;
};

export type OpenAITextFormat = {
  type: string;
};

export type OpenAIText = {
  format: OpenAITextFormat;
};

export type OpenAIUsageDetails = {
  cached_tokens?: number;
};

export type OpenAIOutputTokensDetails = {
  reasoning_tokens?: number;
};

export type OpenAIUsage = {
  input_tokens: number;
  input_tokens_details?: OpenAIUsageDetails;
  output_tokens: number;
  output_tokens_details?: OpenAIOutputTokensDetails;
  total_tokens: number;
};

export type OpenAIResponse = {
  id: string;
  object: string;
  created_at: number;
  status: string;
  background?: boolean;
  error?: any;
  incomplete_details?: any;
  instructions?: any;
  max_output_tokens?: any;
  max_tool_calls?: any;
  model: string;
  output: OpenAIOutputMessage[];
  parallel_tool_calls?: boolean;
  previous_response_id?: string | null;
  reasoning?: OpenAIReasoning;
  service_tier?: string;
  store?: boolean;
  temperature?: number;
  text?: OpenAIText;
  tool_choice?: string;
  tools?: any[];
  top_logprobs?: number;
  top_p?: number;
  truncation?: string;
  usage?: OpenAIUsage;
  user?: any;
  metadata?: { [key: string]: any };
};

File: backend/models/openai_response.py
from pydantic import BaseModel, Field
from typing import Optional, List, Dict, Any

class Document(BaseModel):
    filename: str
    checksum_sha256: str
    size_bytes: int
    storage_key: str
    content_type: str
    created_at: str
    download_url: str

class OpenAIMessage(BaseModel):
    role: str
    content: str

class OpenAIChatRequest(BaseModel):
    model: str
    messages: List[OpenAIMessage]
    documents: Optional[List[Document]] = Field(default_factory=list, description="List of documents to search in Qdrant")
    max_results: Optional[int] = Field(default=5, ge=1, le=20, description="Maximum number of results to return from Qdrant search")

class OpenAIContentItem(BaseModel):
    type: str
    annotations: Optional[List[Any]] = None
    logprobs: Optional[List[Any]] = None
    text: Optional[str] = None

class OpenAIOutputMessage(BaseModel):
    id: str
    type: str
    status: str
    content: List[OpenAIContentItem]
    role: str

class OpenAIReasoning(BaseModel):
    effort: Optional[Any] = None
    summary: Optional[Any] = None

class OpenAITextFormat(BaseModel):
    type: str

class OpenAIText(BaseModel):
    format: OpenAITextFormat

class OpenAIUsageDetails(BaseModel):
    cached_tokens: Optional[int] = None

class OpenAIOutputTokensDetails(BaseModel):
    reasoning_tokens: Optional[int] = None

class OpenAIUsage(BaseModel):
    input_tokens: int
    input_tokens_details: Optional[OpenAIUsageDetails] = None
    output_tokens: int
    output_tokens_details: Optional[OpenAIOutputTokensDetails] = None
    total_tokens: int

class OpenAIResponse(BaseModel):
    id: str
    object: str
    created_at: int
    status: str
    background: Optional[bool] = None
    error: Optional[Any] = None
    incomplete_details: Optional[Any] = None
    instructions: Optional[Any] = None
    max_output_tokens: Optional[Any] = None
    max_tool_calls: Optional[Any] = None
    model: str
    output: List[OpenAIOutputMessage]
    parallel_tool_calls: Optional[bool] = None
    previous_response_id: Optional[str] = None
    reasoning: Optional[OpenAIReasoning] = None
    service_tier: Optional[str] = None
    store: Optional[bool] = None
    temperature: Optional[float] = None
    text: Optional[OpenAIText] = None
    tool_choice: Optional[str] = None
    tools: Optional[List[Any]] = None
    top_logprobs: Optional[int] = None
    top_p: Optional[float] = None
    truncation: Optional[str] = None
    usage: Optional[OpenAIUsage] = None
    user: Optional[Any] = None
    metadata: Optional[Dict[str, Any]] = None
